### DISCUSSION TITLE
Element Relationship field

### DISCUSSION THREAD

**Discussion by Michael-Paragonn** [craftcms/cms#15976] [Ideas]
We have Entry fields, Category fields, Tag fields, etc etc... but what if we need to create a relationship between multiple of those types in one field? It would be great if we could have an **Element Relationship** field, which would allow developers to choose both Sections, Category Groups, etc. for content authors to establish a relationship with.

For example, a site I'm working on has a "You might be interested in" slideshow section at the bottom of their Case Studies. They would like to be able to link to other Case Studies (entries), _as well as_ certain Categories in that slideshow—they have Case Studies categorized by "Services" and by "Industry", and the URL of each category in those is its own landing page.

Having both Entries and Categories selectable from _one_ relationship field would really simplify things in this case. I think it also makes sense, given that Craft is moving towards an "everything is an element" approach these days.

**Comment by thupsi** [comment]
Could you maybe use the new link field for this? Or is it missing some functionality you need?

**Comment by Michael-Paragonn** [comment]
Weeellllll.... it's a Craft 4 site. 😅 So I don't actually know yet; I'm about to begin working on my first Craft 5 site imminently, but this client's site is not that. ¯\\\_(ツ)_/¯

I will say that in this instance, all I'd need is the ability to select various elements from various sources; if the link field also shows fields for the link text, an "Open in new tab" checkbox, etc — all that's extraneous to this use-case.

Can the new Link field return linked-to elements AS elements? I.e. I can continue to query them to fetch their custom Main Image and Tag Line fields? No difference in performance doing that via the Link field vs via a Relation field?

**Reply by thupsi** [reply]
- There is no link text, nor an "Open in a new tab" option. The link field looks like this: 
![image](https://github.com/user-attachments/assets/dbc7c14f-ff8f-42dd-a0fc-e896995a5446)

- Haven't had the chance to test it myself, but judging from the [documentation](https://craftcms.com/docs/5.x/reference/field-types/link.html) it seems that link fields can return linked-to elements AS elements like this: `{{ entry.myLinkFieldHandle.element }}`, so you can query them to fetch any custom fields.
- The docs have a  warning that "It is not currently possible to [eager-load](https://craftcms.com/docs/5.x/development/eager-loading.html) link field relationships, as their element type is not known until the field data is loaded." So there _is_ a performance difference vs a relation field. I suspect though, that an "element relation" field like the one you are suggesting (and I would like to note here that I like the idea), would pose the same difficulty when it comes to eager loading.
- Another point that you may care about is that the link field currently only shows a _chip_ of the linked element, so no _card view_.

**Reply by thupsi** [reply]
Hmm, another important point that I just thought about is that a link field can only contain a single link, so to emulate a relational field with multiple relations you would need a matrix field and have the link field as an entry type in that.

**Reply by thupsi** [reply]
If you don't care about _linking to assets_ though, you could just entrify everything and use an entries field? As you said, since categories and tags are going away eventualy, that could be a good solution going forward, and you can do it in Craft 4. The only frustration is that currently the UI in the CP hasn't quite caught up with the whole "everything is an entry" philosophy.

**Reply by Michael-Paragonn** [reply]
> Hmm, another important point that I just thought about is that a link field can only contain a single link, so to emulate a relational field with multiple relations you would need a matrix field and have the link field as an entry type in that.

Ah, that's a pretty big point for this use-case. However, your idea of using the Link field _did_ lead me to exploring using Hyper (which was already installed on the site) in the same way as you suggested, and _it_ supports multiple links per field! :)

So while I still think an Element Relation field makes sense, my specific need for it has been absolved by the 3rd-party plugin. ¯\\\_(ツ)_/¯
